### ISSUE TITLE
[SURE-7047][v0.9] Support Azure Devops for webhooks Fleet

### ISSUE THREAD

**Issue by kkaempf** [rancher/fleet#1997]
### Issue description:
Fleet currently supports Github, GitLab, Bitbucket, Bitbucket Server and Gogs, https://fleet.rancher.io/webhook.

### Business impact:
Customer can not use webhooks for pulling Git repos, as they use AzureDevops

### Troubleshooting steps:

*    Deploy Rancher Upstream cluster 2.7.5, rke2, 1.24
*    Deploy 2 downstream clusters: RKE2 and EKS
*    Create Azure Devops project (https://dev.azure.com/mrolmedo/_git/fleetproject)
*    Set up webhooks for Azure Devops following this doc: https://learn.microsoft.com/en-us/azure/devops/service-hooks/services/webhooks?view=azure-devops
*    Create a Git Repo in Rancher upstream to deploy some example applications.
*    The webhook fails with error: 
*    Error in logs while sending/updating charts: "2023-10-06T11:40:43Z" level=debug msg="Ignoring unknown webhook event"

**Comment by kkaempf** [comment]
Comment in SURE-7047: "Adding Azure shouldn't be too hard, since there is support in version 6 (https://github.com/go-playground/webhooks/releases/tag/v6.3.0) of the library Gitjob uses."

**Comment by raulcabello** [comment]
# QA Template

## Solution
Add webhook support for Azure Webhooks. It works the same way as existing webhooks for `GitHub`, `Gitlab` and `BitBucket`

## Testing
- Verify webhook in azure are working when no basic authentication is provided
- Wrong basic auth credentials should show an error, and it should not update the Bundle
- Valid auth credentials should work fine
- Webhooks for other providers (`GitHub`, `Gitlab` and `BitBucket`) should keep working

## Additional notes
- How to create Azure Devops Webhooks https://learn.microsoft.com/en-us/azure/devops/service-hooks/services/webhooks?view=azure-devops-2020
- How to configure webhooks in fleet https://fleet.rancher.io/webhook
- Make sure the `syncInterval` is set to something very high (e.g 10000), so it does not interfere with the webhook tests

**Comment by thehejik** [comment]
### Test report
Works as expected.

#### Environment
Tested on local k3s with 1.27 and downstream k3s with `v1.28.6+k3s2` on `rancher:v2.8.3-rc1` with fleet App `103.1.1+up0.9.1-rc.5` using following images:
```
rancher/fleet:v0.9.1-rc.5
rancher/gitjob:v0.9.3
rancher/fleet-agent:v0.9.1-rc.5
```

#### Testing
1) Added an Azure DevOps private HTTPS GitRepo into both workspaces (fleet-local, fleet-default) and with "disabled" polling via `spec.pollingInterval: 48h` set for GitRepos.

2) Created webhook secret for Azure:
```bash
kubectl create secret generic gitjob-webhook -n cattle-fleet-system \
  --from-literal=azure-username=<username> \
  --from-literal=azure-password=<password>
```
3) Made a change in Azure repo and pushed to origin, observing output on local cluster:
```bash
kubectl logs -n cattle-fleet-system -lapp=gitjob -f
```

* With wrong password set on Azure webhook - no change noticed
```
time="2024-02-28T15:05:19Z" level=debug msg="Webhook payload <nil>"
time="2024-02-28T15:05:19Z" level=error msg="Webhook processing failed: basic auth verification failed"
```

* With correct password set - change noticed and resources updated in both workspaces:
```
time="2024-02-28T15:13:37Z" level=debug msg="Webhook payload {..."
time="2024-02-28T15:13:37Z" level=debug msg="Enqueueing gitjob fleet-default/azure-private-webhook in 172800 seconds"
time="2024-02-28T15:13:37Z" level=debug msg="Enqueueing gitjob fleet-local/azure-private-https in 172800 seconds"
time="2024-02-28T15:13:37Z" level=debug msg="DesiredSet - Created batch/v1, Kind=Job fleet-local/azure-private-https-201ca for sync-repo fleet-local/azure-private-https"
time="2024-02-28T15:13:37Z" level=debug msg="DesiredSet - Created batch/v1, Kind=Job fleet-default/azure-private-webhook-201ca for sync-repo fleet-default/azure-private-webhook"
time="2024-02-28T15:13:37Z" level=debug msg="Job change, syncing status for gitjob fleet-default/azure-private-webhook: InProgress"
time="2024-02-28T15:13:37Z" level=debug msg="Job change, syncing status for gitjob fleet-local/azure-private-https: InProgress"
...
time="2024-02-28T15:13:43Z" level=debug msg="Job change, syncing status for gitjob fleet-local/azure-private-https: Current"
time="2024-02-28T15:13:43Z" level=debug msg="Enqueueing gitjob fleet-local/azure-private-https in 172800 seconds"
time="2024-02-28T15:13:43Z" level=debug msg="Job change, syncing status for gitjob fleet-default/azure-private-webhook: Current"
time="2024-02-28T15:13:43Z" level=debug msg="Enqueueing gitjob fleet-default/azure-private-webhook in 172800 seconds"
```

**Comment by thehejik** [comment]
I guess this can be closed, follow up issue opened.

**Comment by thehejik** [comment]
Closing as implemented and verified in v2.8.3-rc2.
